File: app/src/main/java/com/fittrack/app/data/entity/RoutineExercise.kt
package com.fittrack.app.data.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "routine_exercises",
    foreignKeys = [
        ForeignKey(
            entity = Routine::class,
            parentColumns = ["id"],
            childColumns = ["routineId"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Exercise::class,
            parentColumns = ["id"],
            childColumns = ["exerciseId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [
        Index("routineId"),
        Index("exerciseId")
    ]
)
data class RoutineExercise(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val routineId: Long,
    val exerciseId: Long,
    val orderIndex: Int,
    val suggestedSets: Int,
    val suggestedReps: String // e.g. "8-12"
)

File: ios/FitTrack/FitTrack/Models/RoutineExercise.swift
import Foundation
import SwiftData

@Model
final class RoutineExercise {
    var exercise: Exercise?
    var orderIndex: Int
    var suggestedSets: Int
    var suggestedReps: String
    var routine: Routine?

    init(exercise: Exercise, orderIndex: Int, suggestedSets: Int, suggestedReps: String) {
        self.exercise = exercise
        self.orderIndex = orderIndex
        self.suggestedSets = suggestedSets
        self.suggestedReps = suggestedReps
    }
}
